src/documents/ConnectionMode.ts

```diff
@@ -0,0 +1,5 @@
+/** Determines the connection behavior of the CosmosClient. Note, we currently only support Gateway Mode. */
+export enum ConnectionMode {
+  /** Gateway mode talks to a intermediate gateway which handles the direct communicationi with your individual partitions. */
+  Gateway = 0
```

Comment:
Why is this a numeric enum? Typescripts many ways of doing enums confuse me.

Reply:
This is a numeric enum because that's what it was in v1 and we haven't done much implementation changes under the hood.

![image](https://user-images.githubusercontent.com/2252976/43014950-df866944-8c02-11e8-9e17-ccf62d5e0130.png)

I'm open to changing the enum value, but we should do that in a separate PR that addresses the other implementation issues.